### ISSUE TITLE
Equivalent of "git clone --branch"?

### ISSUE THREAD

**Issue by tomqwpl** [go-git/go-git#413]
I'm trying to understand how I would replicate the behaviour of "git clone --branch" in go-git, perhaps you can help?

In go-git, when you clone, you have to pass a ref string such as "refs/heads/<branch>" or "refs/tags/<tag>". 
With "git clone --branch" you just pass the branch name or the tag name and git then queries the refs on the remote in order to find a match, and then clones that ref. 

Any suggestions?

Can I do a "bare" clone (apologies if my terminology isn't right here), query it to find the refs, find the relevant one, then checkout that ref? Would that be equivalent, without loss of efficiency (I'm trying to avoid cloning head of master and then swapping to the relevant ref afterwards)? If so, any pointers on how to do that?

Thanks for any help.

**Comment by n0madic** [comment]
```go
CloneOptions.ReferenceName = plumbing.ReferenceName("refs/heads/branch-name")
```

**Comment by AriehSchneier** [comment]
Fixed in #755, you should now be able to do something like:
```
	r, err := git.Clone(memory.NewStorage(), memfs.New(), &git.CloneOptions{
		URL: "url",
		ReferenceName: plumbing.ReferenceName("branchName"),
		SingleBranch:  true,
	})
```

**Comment by AriehSchneier** [comment]
@pjbgf This can probably be closed as fixed now
